File: MedicalConnect_front/MedicalConnect/MedicalConnect/Users/Doctor/Models/DoctorEvaluationData.swift
//
//  DoctorEvaluationData.swift
//  MedicalConnect
//
//  Created by Dulce Itamar Vigueras Ballesteros on 28/07/24.
//

import Foundation

struct DoctorEvaluationData: Hashable, Identifiable, Codable {
    let id: UUID = UUID()
    var averageOfEvaluations: Float?
    var numberOfConsults: Int?
    var reviews: [Review]?
}

File: MedicalConnect_back/src/main/java/com/c19_136_swift/MedicalConnect/domain/doctor/models/DoctorEvaluationData.java
package com.c19_136_swift.MedicalConnect.domain.doctor.models;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;
import java.util.UUID;

@Entity(name = "DoctorEvaluationData")
@Table(name = "doctor_evaluation_data")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class DoctorEvaluationData {

    @Id
    private UUID uuid;
    private Float averageOfEvaluations;
    private Integer numberOfConsults;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "doctor_id")
    private Doctor doctor;

    @OneToMany(fetch = FetchType.LAZY,cascade = CascadeType.ALL, mappedBy = "evaluationDoctor")
    private List<Review> reviews;
}
